packages/@lwc/template-compiler/src/__tests__/fixtures/expression-complex/compatibility/html-entities/expected.js

```diff
@@ -0,0 +1,24 @@
+import _implicitStylesheets from "./html-entities.css";
+import _implicitScopedStylesheets from "./html-entities.scoped.css?scoped=true";
+import { freezeTemplate, parseFragment, registerTemplate } from "lwc";
+const $fragment1 = parseFragment`<p${3}>foo&amp;bar</p>`;
+const $fragment2 = parseFragment`<p${3}>const { foo } = bar;</p>`;
```

Comment:
It's weird that we don't keep this as `&#123; foo &#125;`, but I guess it's equivalent.

Can we also add a test with `enableStaticContentOptimization` set to `false`? it goes through a different code path at runtime (no `parseFragment`).